### PR TITLE
Make Consumption Profiles report usable with facility-level data

### PR DESCRIPTION
**Pull request by gcapalbo**
The first few commits are just some refactoring to make this report code a little more readable and extensible. The rest of the PR makes it possible to view facility-level data with this report.

(Warehouse-related changes will come in another PR)

Best reviewed commit by commit
@czue

### CODE REVIEW

File: logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py [hunk 1]
```diff
@@ -60,60 +62,118 @@ def consumption_row(sp, p, datespan):
...
+        extra_table_params = extra_table_params or {}
+        table = {
             "is_datatable": False,
             "is_downloadable": True,
-            "header": table_headers,
-            "data": [consumption_row(sp, p, request.datespan) for p in Product.objects.all()]            
+            "header": self.get_consumption_profile_table_headers(request),
+            "data": [
+                consumption_row(supply_point, p, request.datespan)
+                for p in Product.objects.filter(type__base_level=request.base_level)
```
Comment (gcapalbo) [inline logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py]: looks like this has always been showing inactive products in this report. is that intentional?

File: logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py [hunk 2]
```diff
@@ -87,18 +87,34 @@ def get_consumption_profile_table(self, request, supply_point, extra_table_param
...
+            if is_country(supply_point):
+                location_type = "national"
+            elif is_district(supply_point):
+                location_type = "district"
+            else:
+                raise config.BaseLevel.InvalidReportingSupplyPointException(supply_point.code)
+        else:
+            raise config.BaseLevel.InvalidBaseLevelException(supply_point.code)
+
+        return location_type
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py]: nit but i think you could remove the variable and just make everything a `return`

File: logistics_project/apps/malawi/warehouse/warehouse_view.py [hunk 3]
```diff
@@ -35,11 +36,22 @@ def get_reporting_supply_point(self, request):
...
-        Only needs to be implemented when automatically_adjust_datespan is
+        This method is only called when automatically_adjust_datespan is
         overridden to be True.
+
+        This can be overriden and should return the minimum start date that
+        should be used with this report.
+
+        Note that running min() queries on a large table can be an expensive
+        operation, so looking up the values one time and hard-coding them can
+        be a suitable option.
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/warehouse_view.py]: good thought to just hard code these

File: logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py (hunk 1 above)
Comment (czue) [inline logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py]: i'm not sure it's intentional but there may not be any inactive products in the system.

**Pull request merged**